test/test_functional.py

```diff
@@ -84,7 +84,7 @@ def hello():
 
     pre_size = len(hello.__dict__)
     keys = set(hello.__dict__.keys())
-    data_hello = data("test_data_dict.json")(hello)
+    data_hello = file_data("test_data_dict.json")(file_data("test_other_data.json")(hello))
```

Comment:
Ah, weird that this was `data`.
Same comment applies as above though, can we keep a non-cartesian test separate from a cartesian one?